### DISCUSSION TITLE
Hide comment should hide conversation

### DISCUSSION THREAD

**Discussion by marwie** [community/community#59323] [Discussions]
### Select Topic Area

Product Feedback

Hello,

### current behaviour
when marking a  top-level comment in a github discussion as resolved (via `Hide`) only the selected comment gets hidden.

### expected/desired behaviour
when marking a top-level comment in a github discussion as resolved all the child comments should also be hidden

### screenshots

![image](https://github.com/community/community/assets/5083203/271fefea-0c9e-465a-aeb7-a72a5441b787)

![image](https://github.com/community/community/assets/5083203/9f319374-ce68-4e46-86b8-bbeba1b3640b)

![image](https://github.com/community/community/assets/5083203/4bc6f9a1-06f5-4dd1-b5c8-626d67e8cf82)

**Comment by SamSunshine** [comment]
It appears that you expect the child comments of a top-level comment to be hidden automatically when the parent comment is marked as resolved (via Hide) in GitHub discussions.

Currently, GitHub discussions do not have built-in functionality to automatically hide child comments when the parent comment is marked as resolved. The "Hide" feature is designed to specifically hide the selected comment, and it does not cascade to hide its child comments.

**Reply by marwie** [reply]
Hello, yes that was my expectation and the reason I opened this discussion :)

**Reply by GhostOf0days** [reply]
It wouldn't be hard to hide child comments. Very simple recursion. I think it's another reason related to concerns about the first person providing false information, etc.

**Reply by SamSunshine** [reply]
Hi, got it! :) Here's an AI generate suggestion: 

To hide child comments in GitHub discussions using recursion, you can write a custom script or use a browser extension that modifies the page's CSS or JavaScript. Here's an example of how you could implement this using JavaScript:

1. Open the GitHub discussion page where you want to hide child comments.
2. Open the browser console by right-clicking on the page and selecting "Inspect" or by pressing Ctrl+Shift+J (Windows/Linux) or Command+Option+J (Mac).
3. In the console, paste the following JavaScript code:

```
function hideChildComments() {
  // Select all comment threads
  const commentThreads = document.querySelectorAll('.js-comment-container');

  // Iterate over each comment thread
  for (const thread of commentThreads) {
    // Get the depth of the comment thread
    const depth = thread.dataset.commentDepth;

    // Check if the comment thread is a child comment
    if (depth > 0) {
      // Hide the child comment
      thread.style.display = 'none';
    }
  }
}

// Call the function to hide child comments
hideChildComments();
```

This script will select all comment threads on the page and check their depth. If a comment thread has a depth greater than 0, indicating that it's a child comment, it will hide that thread by setting its display style property to 'none'.

Note that this approach relies on the current structure and class names of GitHub's discussion page. If GitHub updates their page structure, this script may not work anymore.

Also, keep in mind that modifying the behavior of a website using custom scripts or browser extensions may violate the website's terms of service. Use this script responsibly and ensure that you comply with GitHub's terms of service.

**Reply by GhostOf0days** [reply]
Please never answer someone else's question using AI. Not only is it not genuine, but it is often inaccurate. This is an inaccurate situation. Refreshing the webpage after changing stuff in the developer console would return it to normal. The content is served from the server, and changing stuff in the developer console only changes the local copy. That server provides a new copy when the page is reloaded. Again, please never answer someone else's question again using AI. It is not only not genuine but also inaccurate. It is better to let someone else answer than you answer using AI.

**Reply by SamSunshine** [reply]
My intent was to try and use resources openly available to help. I am new to entering the Discussion chats here on GitHub and I appreciate the etiquette lesson. Message well received.

**Comment by roke-julian-lockhart** [comment]
Anecdotally, I've encountered situations in which this hasn't been desirable. They're rare, though.
